model/lifecycle.go

```diff
@@ -59,7 +59,7 @@ func SetVersionActivation(versionId string, active bool, caller string) error {
 
 // SetBuildActivation updates the "active" state of this build and all associated tasks.
 // It also updates the task cache for the build document.
-func SetBuildActivation(buildId string, active bool, caller string) error {
+func SetBuildActivation(buildId string, active bool, caller string, abortInProgress bool) error {
```

Comment:
I might be inclined to not modify this interface, and instead just call both methods from the dbconnector? or have two dbconnctor methods that the route calls? 

I worry about the implications of having two queries in one model method, particularly wrt to atomicity implications, and it's clearer, maybe to have them seperate. Also these are sort of the gnarliest parts of the system, and therefore the riskyest, so breaking them apart more is probably the goal.

Reply:
I split these up in the route